### PR TITLE
Fixed code review stuff

### PR DESCRIPTION
**Pull request by danpersa**

### CODE REVIEW

File: src/main/scala/org/zalando/spearheads/innkeeper/oauth/OAuthService.scala [hunk 1]
```diff
@@ -42,14 +42,9 @@ class OAuthService @Inject() (val config: Config,
 
     Try {
       val json = Await.result(futureJson, 1.second)
+      logger.debug(s"The OAuth says: $json")
```
Comment (mo-gr) [inline src/main/scala/org/zalando/spearheads/innkeeper/oauth/OAuthService.scala]: we went from warn to debug?

File: src/main/scala/org/zalando/spearheads/innkeeper/oauth/OAuthService.scala [hunk 2]
```diff
@@ -42,14 +42,9 @@ class OAuthService @Inject() (val config: Config,
...
       json.parseJson.convertTo[AuthorizedUser]
-    } match {
-      case Success(value) => Some(value)
-      case Failure(ex) => {
-        logger.warn("The OAuth exception is", ex)
-        None
-      }
-    }
+    }.toOption
```
Comment (mo-gr) [inline src/main/scala/org/zalando/spearheads/innkeeper/oauth/OAuthService.scala]: This is not functionally equivalent anymore. Previously, you were logging exceptions. Now you are logging the success case and swallow exceptions. Is that on purpose?

**Comment by mo-gr** [comment]
The move from lists to sets looks very nice 👍🏻

File: src/main/scala/org/zalando/spearheads/innkeeper/oauth/OAuthService.scala (hunk 2 above)
Comment (danpersa) [inline src/main/scala/org/zalando/spearheads/innkeeper/oauth/OAuthService.scala]: yes... usually we don't care about this OAuth exception... getting an OAuth error is the normal use case if the token is not valid...

File: src/main/scala/org/zalando/spearheads/innkeeper/oauth/OAuthService.scala (hunk 1 above)
Comment (danpersa) [inline src/main/scala/org/zalando/spearheads/innkeeper/oauth/OAuthService.scala]: we only need this while developing...

**Pull request merged**